### ISSUE TITLE
Detailed diff does not mark fully computed properties for recomputation when the resource is replaced

### ISSUE THREAD

**Issue by VenelinMartinov** [pulumi/pulumi-terraform-bridge#2660]
Fully computed properties (like ID) do not get marked as unknown in previews which suggest a replace. This leads to a confusing preview as the value will most likely change.

**Comment by VenelinMartinov** [comment]
I've added an example to the description, thanks for flagging that up.

**Comment by VenelinMartinov** [comment]
Looked up what Opentofu does here: https://github.com/opentofu/opentofu/blob/864aa9d1d629090cfc4ddf9fdd344d34dee9793e/internal/tofu/node_resource_abstract_instance.go#L1054

Looks like they re-plan the change with a nil prior if they detect a replace in order to re-computed computed properties.

**Comment by t0yv0** [comment]
Can the example include the before and after program indicating which state transition this is in?

**Comment by VenelinMartinov** [comment]
Yeah, it is also a regression in PF as it previously did detect these changes and displayed them to the user.

Is it possible that this also affects correctness? What if another resource depends on this property and we do not mark it as possibly Unknown? Would the dependent resource always get replaced too?

**Comment by VenelinMartinov** [comment]
Discussed offline. We decided this is indeed a blocker as it affects the preview of all Computed properties and they have regressed compared to not using detailed diff v2.

It is also a pre-existing bug in SDKv2, so affects quite a few resources.

**Comment by VenelinMartinov** [comment]
Attempted to solve this the same way that opentofu does, by re-planning the value with a nil prior after detecting a replace. Unfortunately, we seem to be running into engine - provider protocol limitations:

For the example above and the following detailed diff:
```
detailedDiff: map[string]interface{}{
	"id":  map[string]interface{}{"kind": "UPDATE"},
	"key": map[string]interface{}{"kind": "ADD"},
}
```

we get the following preview:

```
	pulumiOut: `Previewing update (test):
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:test::project::pulumi:pulumi:Stack::project-test]
    +-testprovider:index/test:Test: (replace)
        [id=test-id]
        [urn=urn:pulumi:test::project::testprovider:index/test:Test::p]
      - id : "test-id"
      + key: "value"
Resources:
    +-1 to replace
    1 unchanged
```

I believe this is because the engine is using the values from old state and new inputs for showing the detailed diff. This is wrong because the new inputs have no values for fully computed properties, so we get a delete instead of an Update to UNKNOWN.

**Comment by VenelinMartinov** [comment]
Found some indication in the engine that this might have been considered before:

https://github.com/pulumi/pulumi/blob/8314293d6e9901c747b0d397eff9a4a0c0e00466/pkg/resource/deploy/step_generator.go#L1218-L1221

**Comment by VenelinMartinov** [comment]
Here is a branch where we do this re-planning: https://github.com/pulumi/pulumi-terraform-bridge/pull/2673 It shows the effect on the previews and the wrong display of deletions of computed resources.

**Comment by VenelinMartinov** [comment]
This is blocked on https://github.com/pulumi/pulumi-terraform-bridge/issues/2281
